spinalcordtoolbox/vertebrae/core.py

```diff
@@ -525,15 +525,15 @@ def label_discs(fname_seg, list_disc_z, list_disc_value, verbose=1):
     for i in range(len(list_disc_z)):
         if list_disc_z[i] < nz:
             try:
-                slices = seg.data[:, :, list_disc_z[i]]
-                cx, cy = [int(x) for x in np.round(center_of_mass(slices)).tolist()]
+                slices = seg.data[:, :, list_disc_z[i] - 1:list_disc_z[i] + 1]
+                cx, cy, cz = [int(x) for x in np.round(center_of_mass(slices)).tolist()]
             except EmptyArrayError:
                 logger.warning("During disc labeling, center of mass calculation failed due to discontinuities in "
                                "segmented spinal cord; please check the quality of your segmentation. Using "
                                "interpolated centerline as a fallback.")
                 interpolated_centerline, _, _, _ = get_centerline(seg)
-                slices = interpolated_centerline.data[:, :, list_disc_z[i]]
-                cx, cy = [int(x) for x in np.round(center_of_mass(slices)).tolist()]
+                slices = interpolated_centerline.data[:, :, list_disc_z[i] - 1:list_disc_z[i] + 1]
+                cx, cy, cz = [int(x) for x in np.round(center_of_mass(slices)).tolist()]
```

Comment:
While cleaning up this PR, I found that the changes above break an existing test. 

I'm not really sure why this change was made, since `cz` isn't used, but I'll investigate further after the ivadomed meeting.